### DISCUSSION TITLE
staletime 0 by default?

### DISCUSSION THREAD

**Discussion by jmbldwn** [TanStack/query#735] [General]
This seems counter-intuitive.  Basically react-query is off by default.

Shouldn't the default be Infinity, and I would override when needed?

Not suggesting a change, just would like to understand the intent behind the default.

**Comment by tannerlinsley** [comment]
All this means is that the background refetching is aggressive. The caching still work fantastically and the garbage collection keeps data around for 5 minutes before clearing it out. If we set it to infinity by default, then queries would never invalidate and refetch in the background unless they were specifically told to do so.

**Reply by tannerlinsley** [reply]
IMO, It's better to err on the side of overfetching and having your data as up to date as possible. The opposite (staleTime: Infinity) is avoiding fetching at all costs and constantly running the risk of seeing outdated data.
